src/games/strategy/triplea/ui/screen/UnitsDrawer.java

```diff
@@ -24,86 +25,102 @@
-          m_uiContext.getUnitImageFactory().getUnitImageWidth() + 2, 3);
+      graphics.fillRect(placementPoint.x - bounds.x - 2,
+          placementPoint.y - bounds.y + uiContext.getUnitImageFactory().getUnitImageHeight(),
+          uiContext.getUnitImageFactory().getUnitImageWidth() + 2, 3);
     }
-    final UnitType type = data.getUnitTypeList().getUnitType(m_unitType);
+    final UnitType type = data.getUnitTypeList().getUnitType(unitType);
     if (type == null) {
-      throw new IllegalStateException("Type not found:" + m_unitType);
+      throw new IllegalStateException("Type not found:" + unitType);
+    }
+    final PlayerID owner = data.getPlayerList().getPlayerID(playerName);
+    final Image img = uiContext.getUnitImageFactory().getImage(type, owner, data, damaged > 0 || bombingUnitDamage > 0, disabled);
+    //If Mode 2 is selected the nation Flag gets drawn below the Unit
+    if(!STATIC_UNITS.contains(type.getName()) && drawUnitNationMode == 2){
+      final Image flag = uiContext.getFlagImageFactory().getFlag(owner);
+      final int xoffset = img.getWidth(null) / 2 - flag.getWidth(null) / 2;//centered flag in the middle
+      final int yoffset = img.getHeight(null) / 2 - flag.getHeight(null) / 4 - 5;//centered flag in the middle moved it 1/2 - 5 down 
+      graphics.drawImage(flag, (placementPoint.x - bounds.x) + xoffset, (placementPoint.y - bounds.y) + yoffset, null);
+    }
+    //This Method draws the unit Image
+    graphics.drawImage(img, placementPoint.x - bounds.x, placementPoint.y - bounds.y, null);
+    //If Mode 1 is selected the nation Flag gets drawn next to the Unit
+    if(!STATIC_UNITS.contains(type.getName()) && drawUnitNationMode == 1){
+      final Image flag = uiContext.getFlagImageFactory().getSmallFlag(owner);
+      final int xoffset = img.getWidth(null) - flag.getWidth(null);//If someone wants to put more effort in this, he could add an algorithm to calculate the real
+      final int yoffset = img.getHeight(null) - flag.getHeight(null);//lower right corner - transparency/alpha channel etc.
+      //This Method draws the Flag in the lower right corner of the unit image. Since the position is the upper left corner we have to move the picture up by the height and left by the width.
+      graphics.drawImage(flag, (placementPoint.x - bounds.x) + xoffset, (placementPoint.y - bounds.y) + yoffset, null);
     }
-    final PlayerID owner = data.getPlayerList().getPlayerID(m_playerName);
-    final Image img = m_uiContext.getUnitImageFactory().getImage(type, owner, data,
-        m_damaged > 0 || m_bombingUnitDamage > 0, m_disabled);
-    graphics.drawImage(img, m_placementPoint.x - bounds.x, m_placementPoint.y - bounds.y, null);
     // more then 1 unit of this category
-    if (m_count != 1) {
+    if (count != 1) {
       final int stackSize = mapData.getDefaultUnitsStackSize();
       if (stackSize > 0) { // Display more units as a stack
```

Comment:
Not sure. If not, and you're inclined, please remove it in a  follow-up PR after we get this one merged.